### ISSUE TITLE
Create environment for a different platform

### ISSUE THREAD

**Issue by superbobry** [conda/conda#6857]
**tl;dr** it looks like `conda env create` does not allow to build an environment for a platform different from the host, i.e. one can only create Linux environments on Linux, macOS on macOS etc. This could be useful for (admittedly marginal) case like that of dask/knit#116.

The `create` subcommand allows specifying a custom set of channels when creating an environment from a YAML file, e.g. 

```bash
conda env create -f environment.yml -c https://repo.continuum.io/pkgs/main/linux-64 - c ...
```

These channels would be used *jointly* with the default channels and, as far as I can tell, there is no way to override this behaviour, since `conda env create` unlike `conda env` lacks the `--override-channels` option. Therefore, the following

```bash
conda install PACKAGE
```

would first look up the PACKAGE in the default channels, likely find it and install the version for the host platform. Interestingly, though, a related [`constructor`](https://github.com/conda/constructor) project manages to overcome this limitation, so in the end building a Linux environment on macOS should be possible somehow, right?

**Comment by mbargull** [comment]
In the general case you wouldn't want to `conda install` or `conda create` with packages from a different platform. There are some corner cases, e.g., `post-link.sh` scripts that can run arbitrary code as a post-installation step. Those scripts could, for example, test on which platform they are running and then run macOS-specific code in your Linux-targeted environment. So what you actually want is to collect some data (e.g., package lists or even the package tarballs) but not install those -- that's also what `constructor` does when it builds the installers.
You can try the following scripts (make sure to use the `--download-only` and `--dry-run` options!):
```bash
#!/bin/bash

set -ex
DEST_SUBDIR=osx-64
DEST_PACKAGES=libcxx
WORK_DIR=work
(
    mkdir -p $WORK_DIR
    cd $WORK_DIR
    # pretend to be on another platform
    export CONDA_SUBDIR=$DEST_SUBDIR
    # specify where to download packages
    export CONDA_PKGS_DIRS=./$DEST_SUBDIR
    # download packages to WORK_DIR
    conda create --download-only --yes --name unused-env-name $DEST_PACKAGES
    # use conda-build to turn WORK_DIR into a Conda channel
    conda index .
    # test if all requested packages are installed from WORK_DIR
    conda create --dry-run --name unused-env-name --channel ./ $DEST_PACKAGES
)
ls -la ./$WORK_DIR
```
I wouldn't assume this is a supported use case and YMMV. This, however, only gives you the packages in a local channel; you'll still need a `conda` installation on the target platform. If you really want a full-fledged installer, you should use `constructor` (if you experience issues with `constructor`, please file an issue at its issue tracker).

**Comment by kalefranz** [comment]
Even before you get to post-link script execution in the unlink-link phase, you will often run into problems extracting linux packages on macOS, because the default macOS filesystem is case-insensitive, and linux packages require case-sensitive file systems.

**Comment by mbargull** [comment]
> default macOS filesystem is case-insensitive

Right, forgot that one... Bad Apple! Bad decisions!
